x-pack/plugins/apm/dev_docs/telemetry.md

```diff
@@ -55,15 +55,9 @@ The mapping for the telemetry data is here under `stack_stats.kibana.plugins.apm
 
 The mapping used there can be generated with the output of the [`getTelemetryMapping`](../common/apm_telemetry.ts) function.
 
-To make a change to the mapping, edit this function, run the tests to update the snapshots, then use the `merge_telemetry_mapping` script to merge the data into the telemetry repository.
+The `schema` property of the `makeUsageCollector` call in the [`createApmTelemetry` function](../server/lib/apm_telemetry/index.ts) contains the output of `getTelemetryMapping`. Pull requests with changes to the schema should automatically notify the Telemetry team so they can update the mapping in the telemetry clusters.
```

Comment:
NIT: In order for the Kibana Telemetry team to be notified, devs need to run `node scripts/telemetry_check.js --fix`. It will update the file containing all the schemas (and owned by the Telemetry team).

At the moment, APM is excluded so it doesn't fail (https://github.com/elastic/kibana/blob/637a0d9f98faa8d72780c6f41cf498aea19a7b30/x-pack/.telemetryrc.json#L7) and listed in this issue as pending to be fixed: https://github.com/elastic/kibana/issues/70180

I tried with removing the exclusion but it fails.
For progress, I'd say we merge this PR and tackle this issue separately 👍

Reply:
It looks like the reason it's failing is because file with the usage collector call is parsed directly so the whole schema has to be there as a literal, whereas we're calling a function `schema: getApmTelemetryMapping()`, which it fails to parse as a valid schema.

Reply:
@smith there are two other issues I've found thus far while trying to resolve the failure: 
1. The schema needs to be what's displayed in the doc (as types, not actual values) and that the usage collector needs to be explicitly typed.
<details>
<summary>Here's the schema I got from your mapping</summary>

```javascript
export const apmTelemetrySchema = {
  agents: {
    dotnet: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    go: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    java: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    'js-base': {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    nodejs: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    python: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    ruby: {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
    'rum-js': {
      agent: {
        version: { type: 'keyword' },
      },
      service: {
        framework: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        language: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
        runtime: {
          composite: { type: 'keyword' },
          name: { type: 'keyword' },
          version: { type: 'keyword' },
        },
      },
    },
  },
  cloud: {
    availability_zone: { type: 'keyword' },
    provider: { type: 'keyword' },
    region: { type: 'keyword' },
  },
  counts: {
    agent_configuration: {
      all: { type: 'long' },
    },
    error: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
    max_error_groups_per_service: {
      '1d': { type: 'long' },
    },
    max_transaction_groups_per_service: {
      '1d': { type: 'long' },
    },
    metric: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
    onboarding: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
    services: {
      '1d': { type: 'long' },
    },
    sourcemap: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
    span: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
    traces: {
      '1d': { type: 'long' },
    },
    transaction: {
      '1d': { type: 'long' },
      all: { type: 'long' },
    },
  },
  cardinality: {
    user_agent: {
      original: {
        all_agents: {
          '1d': { type: 'long' },
        },
        rum: {
          '1d': { type: 'long' },
        },
      },
    },
    transaction: {
      name: {
        all_agents: {
          '1d': { type: 'long' },
        },
        rum: {
          '1d': { type: 'long' },
        },
      },
    },
  },
  has_any_services: {
    type: 'boolean',
  },
  indices: {
    all: {
      total: {
        docs: { count: { type: 'long' } },
        store: { size_in_bytes: { type: 'long' } },
      },
    },
    shards: { total: { type: 'long' } },
  },
  integrations: {
    ml: { all_jobs_count: { type: 'long' } },
  },
  retainment: {
    error: { ms: { type: 'long' } },
    metric: { ms: { type: 'long' } },
    onboarding: { ms: { type: 'long' } },
    span: { ms: { type: 'long' } },
    transaction: { ms: { type: 'long' } },
  },
  services_per_agent: {
    dotnet: { type: 'long' },
    go: { type: 'long' },
    java: { type: 'long' },
    'js-base': { type: 'long' },
    nodejs: { type: 'long' },
    python: { type: 'long' },
    ruby: { type: 'long' },
    'rum-js': { type: 'long' },
  },
  tasks: {
    agent_configuration: { took: { ms: { type: 'long' } } },
    agents: { took: { ms: { type: 'long' } } },
    cardinality: { took: { ms: { type: 'long' } } },
    cloud: { took: { ms: { type: 'long' } } },
    groupings: { took: { ms: { type: 'long' } } },
    indices_stats: { took: { ms: { type: 'long' } } },
    integrations: { took: { ms: { type: 'long' } } },
    processor_events: { took: { ms: { type: 'long' } } },
    services: { took: { ms: { type: 'long' } } },
    versions: { took: { ms: { type: 'long' } } },
  },
  version: {
    apm_server: {
      major: { type: 'long' },
      minor: { type: 'long' },
      patch: { type: 'long' },
    },
  },
};
```

</details>

2. The usageCollector is complaining of a missing `type`, even though `type: "apm"` is there. I _think_ it's because the collector isn't typed explicitly but we'll figure all these issues out in [#70180](https://github.com/elastic/kibana/issues/70180).